src/utils/geoLib.ts

```diff
@@ -0,0 +1,61 @@
+class GeoLib {
+  private apiKey: string;
+  private baseUrl: string;
+
+  constructor() {
+    this.apiKey = process.env.GOOGLE_GEOCODING_API_KEY;
+    this.baseUrl = "https://maps.googleapis.com/maps/api/geocode/json";
+  }
+
+  public async getAddressFromCoordinates(
+    coordinates: [number, number] | { lng: number; lat: number }
+  ): Promise<string> {
+    try {
+      //Ensure that the coordinates comes as [lng,lat] because is the standard format
+      //to work using 2dsphere format for coordinates.
+      const [lng, lat] = Array.isArray(coordinates)
+        ? coordinates
+        : [coordinates.lng, coordinates.lat];
+      const url = `${this.baseUrl}?latlng=${lat},${lng}&key=${this.apiKey}`;
+
+      const response = await fetch(url);
+      const data = await response.json();
+
+      if (data.status === "OK") {
+        return data.results[0].formatted_address; // Returns the full address
+      } else {
+        throw new Error(data.error_message || "Unable to geocode coordinates");
+      }

```

Comment:
se tem return não precisa de else

```suggestion
      }

      throw new Error(data.error_message || "Unable to geocode coordinates");

```